src/mixins/raster-layer-line-mixin.js

```diff
@@ -0,0 +1,869 @@
+     * @param {Number} s Scale from world to object space.
+     */
+    getSvgPath(t, s) {
+      throw new Error("This must be overridden")
+    }
+  }
+
+  function buildGeoProjection(
+    width,
+    height,
+    margins,
+    xscale,
+    yscale,
+    clamp = true,
+    t = [0, 0],
+    s = 1
+  ) {
+    let _translation = t,
+      _scale = s,
+      _clamp = clamp
+
+    const project = d3.geo.transform({
+      point(lon, lat) {
+        const projectedCoord = conv4326To900913(lon, lat)
+        const pt = [
+          _scale * (xscale(projectedCoord[0]) + margins.left - _translation[0]),
+          _scale *
+          (height -
+            yscale(projectedCoord[1]) -
+            1 +
+            margins.top -
+            _translation[1])
+        ]
+        if (_clamp) {
+          if (pt[0] >= 0 && pt[0] < width && pt[1] >= 0 && pt[1] < height) {
+            return this.stream.point(pt[0], pt[1])
+          }
+        } else {
+          return this.stream.point(pt[0], pt[1])
+        }
+      }
+    })
+
+    project.setTransforms = (t, s) => {
+      _translation = t
+      _scale = s
+    }
+
+    project.setClamp = clamp => {
+      _clamp = Boolean(clamp)
+    }
+
+    return project
+  }
+
+  class GeoSvgFormatter extends PolySvgFormatter {
+    constructor(geocol) {
+      super()
+      this._geojson = null
+      this._projector = null
+      this._d3projector = null
+      this._geocol = geocol
+    }
+
+    getBounds(data, width, height, margins, xscale, yscale) {
+      const wkt = data[this._geocol]
+      if (typeof wkt !== "string") {
+        throw new Error(
+          `Cannot create SVG from geo polygon column "${
+            this._geocol
+            }". The data returned is not a WKT string. It is of type: ${typeof wkt}`
+        )
+      }
+      this._geojson = wellknown.parse(wkt)
+      this._projector = buildGeoProjection(
+        width,
+        height,
+        margins,
+        xscale,
+        yscale,
+        true
+      )
+
+      // NOTE: d3.geo.path() streaming requires polygons to duplicate the first vertex in the last slot
+      // to complete a full loop. If the first vertex is not duplicated, the last vertex can be dropped.
+      // This is currently a requirement for the incoming WKT string, but is not error checked by d3.
+      this._d3projector = d3.geo.path().projection(this._projector)
+      const d3bounds = this._d3projector.bounds(this._geojson)
+      return AABox2d.create(
+        d3bounds[0][0],
+        d3bounds[0][1],
+        d3bounds[1][0],
+        d3bounds[1][1]
+      )
+    }
+
+    getSvgPath(t, s) {
+      this._projector.setTransforms(t, s)
+      this._projector.setClamp(false)
+      return this._d3projector(this._geojson)
+    }
+  }
+
+  _layer._displayPopup = function(
+    chart,
+    parentElem,
+    data,
+    width,
+    height,
+    margins,
+    xscale,
+    yscale,
+    minPopupArea,
+    animate
+  ) {
+    let geoPathFormatter = null
```

Comment:
Looks like this function is mostly similar to the poly version too except for some subtle differences with the SVG building. Perhaps we can coalesce some of this code into a utility function?